src/connector/src/sink/print.rs

```diff
@@ -0,0 +1,201 @@
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use itertools::join;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
+
+use crate::sink::{Result, Sink};
+
+pub const PRINT_SINK: &str = "print";
+
+#[derive(Clone, Debug)]
+pub struct PrintConfig {
+    pub prefix: String,
+    pub suffix: String,
+}
+
+impl PrintConfig {
+    pub fn from_hashmap(values: HashMap<String, String>) -> Result<Self> {
+        let prefix = values
+            .get("prefix")
+            .expect("prefix is required")
+            .to_string();
+        let suffix = values
+            .get("suffix")
+            .expect("suffix is required")
+            .to_string();
+        Ok(PrintConfig { prefix, suffix })
+    }
+}
+
+#[derive(Debug)]
+pub struct PrintSink {
+    pub epoch: u64,
+    pub buffer: Vec<String>,
+    pub prefix: String,
+    pub suffix: String,
+}
+
+impl PrintSink {
+    pub async fn new(config: PrintConfig) -> Result<Self> {
+        Ok(PrintSink {
+            epoch: 0,
+            buffer: vec![],
+            prefix: config.prefix,
+            suffix: config.suffix,
+        })
+    }
+
+    fn parse_datum(datum: DatumRef<'_>) -> String {
+        match datum {
+            None => "NULL".to_string(),
+            Some(ScalarRefImpl::Int32(v)) => v.to_string(),
+            Some(ScalarRefImpl::Int64(v)) => v.to_string(),
+            Some(ScalarRefImpl::Float32(v)) => v.to_string(),
+            Some(ScalarRefImpl::Float64(v)) => v.to_string(),
+            Some(ScalarRefImpl::Decimal(v)) => v.to_string(),
+            Some(ScalarRefImpl::Utf8(v)) => v.to_string(),
+            Some(ScalarRefImpl::Bool(v)) => v.to_string(),
+            Some(ScalarRefImpl::NaiveDate(v)) => v.to_string(),
+            Some(ScalarRefImpl::NaiveTime(v)) => v.to_string(),
+            Some(ScalarRefImpl::Interval(v)) => v.to_string(),
+            _ => unimplemented!(),
+        }
+    }
+}
+
+#[async_trait]
+impl Sink for PrintSink {
+    async fn write_batch(&mut self, chunk: StreamChunk, _schema: &Schema) -> Result<()> {
+        for (op, row_ref) in chunk.rows() {
+            let row_repr = join(row_ref.values().map(Self::parse_datum), ",");
+            let op_repr = match op {
+                Op::Insert => "Insert",
+                Op::UpdateDelete => "UpdateDelete",
+                Op::UpdateInsert => "UpdateInsert",
+                Op::Delete => "Delete",
+            };
+            self.buffer.push(format!("{}{}", op_repr, row_repr));
+        }
+
+        Ok(())
+    }
+
+    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
+        self.epoch = epoch;
+        Ok(())
+    }
+
+    async fn commit(&mut self) -> Result<()> {
+        for row in self.buffer.clone() {
+            println!("{}{}{}", self.prefix, row, self.suffix)
+        }
+        self.buffer = vec![];
```

Comment:
```suggestion
        self.buffer.clear();
```

new a vec will drop the memory of the prev one. We are using the buffer later so it is better to use `clear` here, which will not deallocate the memory.